tools/read_collection.cpp

```diff
@@ -4,54 +4,79 @@
 #include <range/v3/view/iota.hpp>
 
 #include "binary_collection.hpp"
+#include "io.hpp"
+#include "memory_source.hpp"
+#include "payload_vector.hpp"
 #include "util/util.hpp"
 
 using namespace pisa;
 
 int main(int argc, char** argv)
 {
     std::string collection_file;
-    std::string map_file{};
+    std::optional<std::string> map_file{};
+    std::optional<std::string> lex_file{};
     std::ptrdiff_t first, last;
 
-    CLI::App app{"read_collection - read collections."};
+    CLI::App app{"Reads binary collection to stdout."};
     app.add_option("-c,--collection", collection_file, "Collection file name")->required();
-    app.add_option("-m,--map", map_file, "String map file name");
-    app.add_option("first", first, "Element number")->required();
-    app.add_option("last", last, "Element number");
+
+    auto maptext =
+        app.add_option("--maptext", map_file, "Text file mapping each line number (ID) to a string");
+    auto maplex = app.add_option("--maplex", lex_file, "Lexicon file mapping IDs to strings");
+    maptext->excludes(maplex);
+    maplex->excludes(maptext);
```

Comment:
I think having both is redundant, but it does not hurt too.

Reply:
It's good for when the help message is printed, whichever you look at, you immediately see it excludes the other.